Update UI to be a subtree of the Antora default UI (#1906)
LiskHQ/lisk-docs: ui/docs/modules/ROOT/pages/create-helper.adoc

= Create a UI Helper

This page explains how to create a UI helper for use in a page template (layout or partial).
A helper is a JavaScript function that's invoked by Handlebars when it comes across a helper call in a template.

== Helper anatomy

A helper must be defined as a JavaScript file in the [.path]_helpers_ directory of the UI bundle.
The basename of the file without the file extension will be used as the function name.
For example, if the helper is located at [.path]_helpers/join.js_, the name of the function will be `join`.

You don't have to register the helper as Antora does that for you automatically.
This automatic behavior replaces this Handlebars API call (which you *don't* have to do):

[,js]
----
Handlebars.registerHelper('join', function () { ... })
----

The helper file should export exactly one default function.
The name of the function in the file does not matter.

> Here's a template of a helper function you can use as a starting point:

Reviewer: ```suggestion
A template of a helper function you can use as a starting point is shown below:
```

.new-helper.js
[,js]
----
'use strict'

module.exports = () => {
  return true
}
----

The return value of the function will be used in the logic in the template.
If the helper is used in a conditional, it should return a boolean value (as in the previous example).

> If the helper is used to create output, it should return a string.

Reviewer: ```suggestion
If the helper is used to create an output, it should return a string.
```

We can now use our conditional helper in a template as follows:

[,hbs]
----
{{#if (new-helper)}}
always true!
{{/if}}
----

The round brackets are always required around a helper function call (except in cases when they're implied by Handlebars).

The helper can access top-level variables in the template by accepting the template context as the final parameter.
The top-level variables are stored in in the `data.root` property of this object.

.new-helper.js
[,js]
----
'use strict'

module.exports = ({ data: { root } }) => {
  return root.site.url === 'https://docs.example.org'
}
----

Now our condition will change:

[,hbs]
----
{{#if (new-helper)}}
Only true if the site URL is https://docs.example.org.
{{/if}}
----

A helper can also accept input parameters.
These parameters get inserted in the parameter list before the context object.
Handlebars only calls the function with the input parameters passed by the template, so it's important to use a fixed number of them.
Otherwise, the position of the context object will jump around.

.new-helper.js
[,js]
----
'use strict'

module.exports = (urlToCheck, { data: { root } }) => {
  return root.site.url === urlToCheck
}
----

> Now we can accept the URL to check as an input parameter:

Reviewer: ```suggestion
Now we can accept the URL to check as an input parameter as shown below:
```
